src/plugins/explorer/explorer.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+const axios = require('axios')
+const EventEmitter = require('events')
+const LumerinContracts = require('@lumerin/contracts')
+
+class Explorer {
+  constructor({ chainId, web3 }) {
+    switch (chainId.toString()) {
+      case 'mainnet':
+      case '1':
+        this.apiUrl = 'https://api.etherscan.io/api'
+        break
+      case 'ropsten':
+      case '3':
+        this.apiUrl = 'https://api-ropsten.etherscan.io/api'
+        break
+      default:
+        throw new Error(`Unsupported chain ${chainId}`)
+    }
```

Comment:
i would suggest creating a separate factory function for that, which will inject apiUrl and Lumerin instance to explorer class dependencies
```
const ExplorerFactory = (chainId, web3): Explorer => {
}
```

Reply:
@lsheva now you like the factory pattern ;)

@alex-sandrk shev's suggestion would make your plugin more consistent with the other plugins I think.